pkg/registry/bundle.go

```diff
@@ -40,15 +56,25 @@ type Bundle struct {
 	Package      string
 	Channels     []string
 	BundleImage  string
-	csv          *ClusterServiceVersion
-	v1beta1crds  []*apiextensionsv1beta1.CustomResourceDefinition
-	v1crds       []*apiextensionsv1.CustomResourceDefinition
 	Dependencies []*Dependency
-	cacheStale   bool
+
+	csv         *ClusterServiceVersion
+	v1beta1crds []*apiextensionsv1beta1.CustomResourceDefinition
+	v1crds      []*apiextensionsv1.CustomResourceDefinition
+	cacheStale  bool
+	// If true, all operations that assume a CSV exists in Bundle must return an
+	// error if one does not.
+	mustHaveCSV bool
 }
 
 func NewBundle(name, pkgName string, channels []string, objs ...*unstructured.Unstructured) *Bundle {
-	bundle := &Bundle{Name: name, Package: pkgName, Channels: channels, cacheStale: false}
+	bundle := &Bundle{
+		Name:        name,
+		Package:     pkgName,
+		Channels:    channels,
+		cacheStale:  false,
+		mustHaveCSV: true,
```

Comment:
I see your reasoning for adding this flag. However, there must be a way to set this flag somewhere so we have control over this flag. It may be true for now but may not be in the future.

Reply:
I am thinking the `Bundle` constructor can set this option in the future via a constructor parameter or inference on `Bundle.Objects`. For now it will always be true. I'm fine with just checking `Bundle.csv == nil` if you aren't sure a boolean option will be used to determine bundle type going forward.